docs: port conditions, IPv6 configuration, known issues, and nodetool alternatives references to staging (#3420)
scylladb/scylla-operator: docs-staging/source/reference/conditions.md

# Conditions reference

The ScyllaDB Operator uses Kubernetes status conditions to communicate health and progress of managed resources.
This page documents the top-level condition types, their semantics, and aggregation rules.

These conditions apply to all Operator-managed custom resources that report status conditions (e.g. `ScyllaCluster`, `ScyllaDBDatacenter`, `NodeConfig`).

## Top-level conditions

Every condition-reporting resource exposes three top-level conditions:

> | Condition type | Meaning when `True` | Meaning when `False` |

Reviewer: Should we document `Unknown` as well? https://github.com/scylladb/scylla-operator/blob/5c8ca6bf07806c363d0a3b0662033d90c0810146/pkg/controllerhelpers/core.go#L194

Applies to line 48 as well.

Author: Documented `Unknown`.

|---|---|---|
| `Available` | The resource is functional and can serve its purpose | The resource is not functional |

> | `Progressing` | The Operator is actively reconciling (rolling update, scaling, version change) | No active reconciliation in progress |

Reviewer: ```suggestion
| `Progressing` | The Operator is actively reconciling (rolling update, scaling, version change) | Either completed or stuck |
```

Author: Applied.

> | `Degraded` | Something is unhealthy or an error occurred during reconciliation | Everything is healthy and reconciliation is error-free |

Reviewer: ```suggestion
| `Degraded` | Something is unhealthy or an error occurred during reconciliation | Resource is healthy |
```

Author: Applied.
